src/sentry/search/events/fields.py

```diff
@@ -1869,13 +1907,21 @@ def resolve_field(self, field: str) -> Union[SnqlColumn, SnqlFunction]:
     @property
     def orderby(self) -> List[OrderBy]:
         validated = []
-        for column in self.orderby_columns:
-            bare_column = self.column(column.lstrip("-"))
-            direction = Direction.DESC if column.startswith("-") else Direction.ASC
-
-            if bare_column in self.columns:
-                validated.append(OrderBy(bare_column, direction))
-                continue
+        for orderby in self.orderby_columns:
+            bare_orderby = orderby.lstrip("-")
+            resolved_orderby = self.column(bare_orderby)
+            direction = Direction.DESC if orderby.startswith("-") else Direction.ASC
+
+            for selected_column in self.columns:
+                if isinstance(selected_column, SnqlColumn) and selected_column == resolved_orderby:
+                    validated.append(OrderBy(selected_column, direction))
+                    continue
+                elif (
+                    isinstance(selected_column, SnqlFunction)
+                    and selected_column.alias == bare_orderby
+                ):
+                    validated.append(OrderBy(selected_column, direction))
+                    continue
```

Comment:
Nit: this nested loop is now `O(n^2)`, can this be flattened with 2 sets, one with the `SnqlColumn`s and `SnqlFunction`?